docs/designs/ruby_test_plan_002.md

```diff
@@ -0,0 +1,150 @@
+demonstrating an aspect under test. A list of necessary repositories could be
+found [below](#necessary-repositories).
+
+## Test cases
+
+@e2e_test
+Feature: Projects prepared with Cachi2 could be hermetically built
+    Scenario: a user can successfully build a project prepared with Cachi2
+        Given a clean repository of a well-defined project
+         When a user request to fetch dependencies for a well-defined project
+          And a user generates environment for well-defined project
+          And a user triggers a hermetic build
+         Then the build system exits successfully
+          And a build artifact is produced.
+
+
+@integration_test
+Feature: Cachi2 can prefetch dependencies for Ruby projects
+    
+    Scenario outline: a user wants to fetch dependencies for a package containing well-defined Gemfile.lock
+       Given  clean repository of a "well-defined project"
+        When  a user request to fetch dependencies for a "well-defined project"
+        Then  direct and transitive dependencies are fetched
+         And  dependencies could be found within the "well-defined project"
+    Examples:
+        |    well-defined project                          |
+        | Depends on gems from rubygems.org only           |
+        | Has correct .gemspec                             |
+        | Has correct Git dependency                       |
+        | Has several correct Git dependencies             |
+        | Has path dependency                              |
+        | Has several path dependencies                    |
+        | Has a mixture of gems, Git and path dependencies |
+
+    Scenario outline: a user wants to fetch dependencies for a package containing ill-defined Gemfile.lock
+       Given  clean repository of a "ill-defined project"
+        When  a user request to fetch dependencies for a "ill-defined project"
+        Then  Cachi2 reports an error
+         And  repository for "ill-defined project" remains unchanged
+    Examples:
+        |    ill-defined project                     |
+        | Gemfile.lock is missing                    |
+        | Gemfile.lock contains critical code        |
+        | Git dependency is not pinned to hash       |
+        | Gem dependency is sourced from unknown URL |
+        | Transitive dependency is ill-defined       |
+    Note: "unknown URL" is any URL other than https://rubygems.org
+
+    Scenario outline: a user wants to fetch dependencies for a package containing poorly-defined Gemfile.lock
+        When  a user requests to fetch dependnecies for a "poorly-defined project"
+        Then  direct and transitive dependencies are fetched
+         And  could be found within the "poorly-defined project"
+         But  Cachi2 warns about detected problem
+    Examples:
+        |    poorly-defined project             |
+        | Gemfile contains executable code      |
+TODO: Should we detect, warn and block packages with any code in Gemfile?
```

Comment:
We are not touching the `Gemfile`, so no problem there. Now, if the question is regarding the `Gemfile.lock`... can arbitrary code be added to it? How does it behave with Bundler and with the gemlock-parser?

My initial guess is that the `Gemfile.lock` is just meant to have structured data, and no code is ever executed there (and the existence of code would make the file invalid).

Reply:
Ok, then this is not a valid example. Removing it. Thank you!